backend/src/schema.graphql

```diff
@@ -25,6 +25,8 @@ type Mutation {
   report(id: ID!, description: String): Report
   disable(id: ID!): ID
   enable(id: ID!): ID
+  reward(fromBadgeId: ID!, toUserId: ID!): ID
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-03-22T10:47:26Z" title="Friday, March 22nd 2019, 11:47:26 am +01:00">Mar 22, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -25,6 +25,8 @@ type Mutation {
   report(id: ID!, description: String): Report
   disable(id: ID!): ID
   enable(id: ID!): ID
+  reward(fromBadgeId: ID!, toUserId: ID!): ID
```


I'm wondering why you return `ID` here :thinking: . To me, it would make sense to return `[Reward]`. So instead of the `id` of the rewarded user you would get the updated set of rewards of that user. This would simplify the test, too :wink: 

@mattwr18 @Tirokk this is an optional change though. I like this PR and it's mergeable, all tests pass, so I'll approve. When we touch the behaviour again (e.g. when implementing frontend part) it would make sense to improve it.

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-03-22T18:13:31Z" title="Friday, March 22nd 2019, 7:13:31 pm +01:00">Mar 22, 2019</time>_

---


I had talked about this in the last pairing session we had together on this @roschaefer 
I didn't make changes to the logic of the `PR`, I just took advantage of `MERGE` in cypher to enforce uniqueness and got the tests passing.

It makes sense to me to return the `Reward` as well although @Kachulio1 was saying maybe we just return a `Boolean`... either way I'm happy to make the change and I'm sure @Tirokk won't have an issue with it

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-03-23T01:33:26Z" title="Saturday, March 23rd 2019, 2:33:26 am +01:00">Mar 23, 2019</time>_

---


`Reward` is a single reward `[Reward]` is a list of rewards. This list could reflect the list after updating. This way, you don't have to re-check this list in the test..